src/state_machine/coordinator/fire.rs

```diff
@@ -658,7 +679,7 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
                                     let signer_key_ids = &self.config.signer_key_ids[signer_id];
 
                                     for (src_party_id, key_shares) in &dkg_private_shares.shares {
-                                        let poly = &dkg_public_shares[src_party_id];
+                                        let poly = &polys[src_party_id];
```

Comment:
Let's change the direct array access here while we are at it.

Reply:
Done.  Fixing this caused a weird Heisenbug that disappeared when I added logging to explore it.  Maybe it was a timing issue?

Reply:
Interesting

Reply:
Yeah it's really weird, and I haven't see it in CI yet.